src/list.c

```diff
@@ -184,11 +184,14 @@ void *list_get_tail(list_t *list) {
 
 /* Fast list deletion */
 
-void list_delete_list(list_t *list) {
+void list_empty_list(list_t *list) {
 	for(list_node_t *node = list->head, *next; next = node ? node->next : NULL, node; node = next) {
 		list_free_node(list, node);
 	}
+}
```

Comment:
You should also set `list->head` and `list->tail` to `NULL` afterwards, and `list->count` to zero.

Reply:
Done.